src/error.rs

```diff
@@ -27,16 +27,21 @@ pub enum Error {
     /// Invalid Fee for conversion
     InvalidFeeConversion,
     /// Poseidon Error
-    PoseidonError,
+    PoseidonError(PoseidonError),
     /// Invalid Value Commitment
     InvalidCommitment,
     /// Invalid Nonce
     InvalidNonce,
 }
 
-impl<E: fmt::Debug> From<PoseidonError<E>> for Error {
-    fn from(_p: PoseidonError<E>) -> Error {
-        // TODO - wrap the concrete error type
-        Error::PoseidonError
+impl From<PoseidonError> for Error {
+    fn from(p: PoseidonError) -> Self {
+        Self::PoseidonError(p)
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Phoenix-Core Error: {:?}", &self)
```

Comment:
I don't think this should be the way to format our errors, but since we do not have a standard yet we can leave as is for the time being.

Reply:
This format is the same of PoseidonError. I agree we should set a better standard.